rfcs/text/0006_management_section_service.md

```diff
@@ -0,0 +1,341 @@
+- Start Date: 2019-08-20
+- RFC PR: TBD
+- Kibana Issue: [#43499](https://github.com/elastic/kibana/issues/43499)
+
+# Summary
+Management is one of the four primary "domains" covered by @elastic/kibana-app-arch (along with Data, Embeddables, and Visualizations). There are two main purposes for this service:
+
+1. Own the management "framework" -- the UI that displays the management sidebar nav, the landing page, and handles rendering each of the sections
+2. Expose a registry for other plugins to add their own registry sections to the UI and add nested links to them in the sidebar.
+
+The purpose of this RFC is to consider item 2 above -- the service for registering sections to the nav & loading them up.
+
+# Motivation
+
+## Why now?
+The main driver for considering this now is that the Management API moving to the new platform is going to block other teams from completing migration, so we need to have an answer to what the new platform version of the API looks like as soon as possible in `7.x`.
+
+## Why not just keep the current API and redesign later?
+The answer to that has to do with the items that are currently used in the management implementation which must be removed in order to migrate to NP: the framework currently registers a `uiExport`, and relies on `IndexedArray`, `uiRegistry`, and `ui/routes`.
+
+This means that we will basically need to rebuild the service anyway in order to migrate to the new platform. So if we are going to invest that time, we might as well invest it in building the API the way we want it to be longer term, rather than creating more work for ourselves later.
+
+## Technical goals
+- Remove another usage of `IndexedArray` & `uiRegistry` (required for migration)
+- Remove dependency on `ui/routes` (required for migration)
+- Remove management section `uiExport` (required for migration)
+- Simple API that is designed in keeping with new platform principles
+  - This includes being rendering-framework-agnostic... You should be able to build your management section UI however you'd like
+- Clear separation of app/UI code and service code, even if both live within the same plugin
+- Flexibility to potentially support alternate layouts in the future (see mockups in [reference section](#reference) below)
+
+# Basic example
+This API is influenced heavily by the [application service mounting RFC](https://github.com/elastic/kibana/blob/master/rfcs/text/0004_application_service_mounting.md). The intent is to make the experience consistent with that service; the Management section is basically one big app with a bunch of registered "subapps".
+
+```ts
+// my_plugin/public/plugin.ts
+
+export class MyPlugin {
+  setup(core, { management }) {
+    // Registering a new link to a new section
+    const mySection = management.sections.register({
+      id: 'my-section',
+      title: 'My Main Section', // display name
+      description: 'hello', // not used in current UI, but possibly in future
```

Comment:
Could I suggest we remove `description` from sections and links if they're not being used?

Reply:
resolved